File: Spring_learn_specification/src/main/java/com/fintech/domain/Person.java
package com.fintech.domain;

import lombok.*;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@EqualsAndHashCode
@Builder
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long Id;
    private String firstName;
    private String lastName;
    private Date dob;
    @Enumerated(value = EnumType.STRING)
    private Status status;

    public enum Status {
        ACTIVE, INACTIVE
    }
}

File: SpringMvcPracticeAnoDataBinderExceptionClient/app/domain/Person.ts
export class Person {
    id : number;
    age : number;
    dob : string;
    firstName :string;
    lastName : string;
    salary : number;
    status : string;
    
    constructor(age: number, dob: string, firstName: string, lastName: string, salary: number, status: string) {
        this.age = age;
        this.dob = dob;
        this.firstName = firstName;
        this.lastName = lastName;
        this.salary = salary;
        this.status = status;
    }
}